be/src/exec/tablet_sink_index_channel.cpp

```diff
@@ -445,14 +459,16 @@ Status NodeChannel::add_chunk(Chunk* input, const std::vector<int64_t>& tablet_i
         std::vector<uint32_t> filtered_indexes;
         RETURN_IF_ERROR(_filter_indexes_with_where_expr(input, indexes, filtered_indexes));
         size_t filter_size = filtered_indexes.size();
-        _cur_chunk->append_selective(*input, filtered_indexes.data(), from, filter_size);
+        _append_data_to_cur_chunk(*input, filtered_indexes.data(), from, filter_size);
+
         auto req = _rpc_request.mutable_requests(0);
         for (size_t i = 0; i < filter_size; ++i) {
             req->add_tablet_ids(tablet_ids[filtered_indexes[from + i]]);
         }
     }
 
-    if (_cur_chunk->num_rows() < _runtime_state->chunk_size()) {
+    if (_cur_chunk->num_rows() < _runtime_state->chunk_size() &&
+        _cur_chunk_mem_usage < config::max_tablet_write_chunk_bytes) {
```

Comment:
_cur_chunk->num_rows() <= 0 || (_cur_chunk->num_rows() < _runtime_state->chunk_size() && _cur_chunk_mem_usage < config::max_tablet_write_chunk_bytes) maybe better

Reply:
clone_empty_with_slot maybe reserve some memory, there maybe no one row in chunk, but memory exceed the limit.